src/GitHubClient.php

```diff
@@ -179,12 +186,29 @@ public function getRepositories(array &$failures = null, $withFundingLinks = fal
     public function call($graphql, array &$failures = [])
     {
         $rfs = $this->rfs;
+        $composer = $this->composer;
 
         if ($eventDispatcher = $this->composer->getEventDispatcher()) {
-            $preFileDownloadEvent = new PreFileDownloadEvent(PluginEvents::PRE_FILE_DOWNLOAD, $rfs, 'https://api.github.com/graphql');
-            $eventDispatcher->dispatch($preFileDownloadEvent->getName(), $preFileDownloadEvent);
-            if (!$preFileDownloadEvent->getRemoteFilesystem() instanceof CurlRemoteFilesystem) {
-                $rfs = $preFileDownloadEvent->getRemoteFilesystem();
+            if (version_compare('2', $composer::getVersion()) < 0) {
```

Comment:
Can we use feature detection instead?

```suggestion
            if (class_exists('Composer\Util\HttpDownloader')) {
```